cmd/kata-local-ci

```diff
@@ -0,0 +1,260 @@
+	    osvariant=ubuntu18.04
+	    ;;
+	centos)
+	    distro_name=centos-8.2
+	    osvariant=centos8
+	    distro_commands=("${rh_distro_commands[@]}")
+	    ;;
+	fedora)
+	    distro_name=fedora-32
+	    osvariant=fedora32
+	    distro_commands=("${rh_distro_commands[@]}")
+	    final_firstboot_cmd=("${disable_cgroupv2[@]}")
+	    ;;
+	*)
+	    echo "Distro $1 not recognized. Please choose ubuntu, centos or fedora"
+            exit 3
+    esac
+
+    if [  ! -z "$network" ]; then
+	network="--network $network"
+    fi
+
+    VM_NAME=kata-$distro-ci
+    progress "Removing earlier VM $VM_NAME"
+    echo "The script may now ask you for a password to setup the VM named $VM_NAME"
+    echo "You can avoid these prompts by executing:"
+    echo "sudo bash -c \"echo > /etc/sudoers.d/kata-local-ci '"$(whoami) "ALL=(ALL) NOPASSWD: /usr/bin/virsh,/usr/bin/virt-install,/usr/bin/virt-builder'\""
+
+    sys virsh destroy $VM_NAME || true
+    sys virsh undefine $VM_NAME --remove-all-storage  --snapshots-metadata || true
+
+    set -e
+    progress "Launching virt-builder"
+    sys virt-builder $distro_name \
+         -o $disk --format qcow2 --size=$size \
+	 --root-password password:kata \
+	 "${distro_commands[@]}" \
+	 --mkdir /home/kata \
+	 --delete /.autorelabel \
+	 --ssh-inject "root:file:$key" \
+	 --install "git,rsync,curl,make,tar,tc" \
+         "${firstboot_cmds[@]}" \
+         "${final_firstboot_cmd[@]}" \
+	 --password  'kata:password:kata' \
+	 --hostname kata-$distro-ci
+    progress "Launching virt-install"
+    sys virt-install --import --name $VM_NAME  --memory $memory --vcpus $cpus \
+         --disk path=$disk,format=qcow2 \
+         --os-variant=$osvariant --noautoconsole $network
+
+    echo -n "Waiting for guest to get an IP address"
+    while ! sudo virsh domifaddr $VM_NAME | grep -q vnet ; do echo -n .; sleep 1; done
+
+    VM_IPADDR=$(sudo virsh domifaddr $VM_NAME | grep vnet | sed -e 's|^\s*vnet[0-9]*.*ipv4\s*\([0-9.]*\)/[0-9]*|\1|g')
+    echo "Got $VM_IPADDR"
+    progress "Guest IP address is $VM_IPADDR"
+
+    progress "Waiting for ssh to become available"
+    while ! ssh -o StrictHostKeyChecking=no -o UpdateHostKeys=no kata@$VM_IPADDR "mkdir -p ~/go/src/github.com/kata-containers"; do
+        sleep 1
+    done
+
+    progress "Initial rsync of kata-containers directory"
+    run rsync -av $directory/kata-containers kata@$VM_IPADDR:$vmdir/
+
+    progress "Initial rsync of tests directory"
+    run rsync -av $directory/tests kata@$VM_IPADDR:$vmdir/
+
+    progress "Launching CI setup in $VM_NAME (this will take a while)"
+    ci_cmd "cd $vmdir/tests; export GOPATH=~/go; .ci/setup.sh"
+
+    cat > "$kci"/$distro-config <<EOF
+VM_NAME="$VM_NAME"
+VM_IPADDR=$VM_IPADDR
+VM_DISK=$disk
+EOF
+
+    progress "Taking snapshot of $VM_NAME"
+    ci_root_cmd "sync; sync; sync; sleep 3"
+    sys virsh snapshot-create-as $VM_NAME --name initial-snapshot --description "Initial snapshot of $VM_NAME after Kata CI setup"
```

Comment:
Worth specifying distro name in the snapshot name?

Reply:
I seem to recall that I dropped it because it's already visible in the VM name, so that was duplicate information.